src/utils/Logger.ts

```diff
@@ -146,15 +148,71 @@ class Logger {
     return error
   }
 
-  getReactNativeLogsFilePath = () => {
-    return RNFS.CachesDirectoryPath + '/rn_logs.txt'
+  getReactNativeLogFilePath = () => {
+    return `${RNFS.CachesDirectoryPath}/rn_logs/${format(new Date(), 'yyyyMMdd')}.txt`
   }
 
-  getLogs = async () => {
-    // TODO(Rossy) Does this technique of passing logs back as a string
-    // fail when the logs get too big?
+  getReactNativeLogsDir = () => {
+    return `${RNFS.CachesDirectoryPath}/rn_logs`
+  }
+
+  getCombinedLogsFilePath = () => {
+    // For now we need to export to a world-readable directory on Android
+    // TODO: use the FileProvider approach so we don't need to do this.
+    // See https://developer.android.com/reference/androidx/core/content/FileProvider
+    // and https://github.com/chirag04/react-native-mail/blame/340618e4ef7f21a29d739d4180c2a267a14093d3/android/src/main/java/com/chirag/RNMail/RNMailModule.java#L106
+    const path = Platform.OS === 'ios' ? RNFS.TemporaryDirectoryPath : RNFS.ExternalDirectoryPath
+    return `${path}/rn_logs.txt`
+  }
+
+  // Initialize the logs directories and files
+  initializeLogs = async () => {
+    const logFileCombinedPath = this.getCombinedLogsFilePath()
+    const logFilePath = this.getReactNativeLogFilePath()
+    const logDir = this.getReactNativeLogsDir()
+    console.debug('React Native logs will be piped to ' + logFilePath)
+
+    // If log folder not present create it
+    if (!(await RNFS.exists(logDir))) {
+      await RNFS.mkdir(logDir)
+    }
+
+    // If daily log file is not present create it
+    if (!(await RNFS.exists(logFilePath))) {
+      await RNFS.writeFile(logFilePath, '', 'utf8')
+    }
```

Comment:
Looks like this is not necessary, right?

Reply:
I was having trouble appending to a file that doesn't exist so I had to check that it was created first.